source/common/statistic_impl.cc

```diff
@@ -9,23 +9,35 @@
 namespace Nighthawk {
 
 std::string StatisticImpl::toString() const {
-  return fmt::format("Count: {}. Mean: {:.{}f} μs. pstdev: {:.{}f} μs.\n", count(), mean() / 1000,
-                     2, pstdev() / 1000, 2);
+  return toProto(SerializationDomain::RAW).DebugString();
 }
 
-nighthawk::client::Statistic StatisticImpl::toProto() {
+nighthawk::client::Statistic StatisticImpl::toProto(SerializationDomain domain) const {
   nighthawk::client::Statistic statistic;
 
   statistic.set_id(id());
   statistic.set_count(count());
-
-  int64_t nanos = count() == 0 ? 0 : static_cast<int64_t>(std::round(mean()));
-  statistic.mutable_mean()->set_seconds(nanos / 1000000000);
-  statistic.mutable_mean()->set_nanos(nanos % 1000000000);
-
-  nanos = count() == 0 ? 0 : static_cast<int64_t>(std::round(std::isnan(pstdev()) ? 0 : pstdev()));
-  statistic.mutable_pstdev()->set_seconds(nanos / 1000000000);
-  statistic.mutable_pstdev()->set_nanos(nanos % 1000000000);
+  if (domain == Statistic::SerializationDomain::DURATION) {
+    int64_t nanos;
+    nanos = count() == 0 ? 0 : static_cast<int64_t>(std::round(mean()));
+    statistic.mutable_mean()->set_seconds(nanos / 1000000000);
```

Comment:
Also since this pattern with /1000000000 and %1000000000, maybe there should be a helper method for setting a proto duration, taking mutable Duration* and int64_t nanos